src/transports/sentry_transport_curl.c

```diff
@@ -17,8 +17,8 @@ typedef struct curl_transport_state_s {
     bool initialized;
     bool debug;
     CURL *curl_handle;
-    const char *http_proxy;
-    const char *ca_certs;
+    char *http_proxy;
+    char *ca_certs;
```

Comment:
All these should be owned by the background task and freed when the task shuts down. As such, how about we move them to their own `sentry_curl_bgw_s` struct and pass it as `data`/`state` parameter into `sentry__bgworker_new()`?